### PR TITLE
Faster direct beam

### PR DESCRIPTION
**Pull request by SimonHeybrock**
Might replace #57. Times are now down to 15 second for the normal iteration (used to be 1-2 minutes).

### CODE REVIEW

File: src/esssans/i_of_q.py [hunk 1]
```diff
@@ -276,9 +278,38 @@ def _dense_merge_spectra(
...
+        ):
+            del stripped.coords[name]
+    to_flatten = [dim for dim in data_q.dims if dim in dims_to_reduce]
+
+    dummy_dim = str(uuid4())
+    flat = stripped.flatten(dims=to_flatten, to=dummy_dim)
+
+    # Apply masks once, to avoid repeated work when iterating over bands
+    mask = irreducible_mask(flat, dummy_dim)
+    # When not all dims are reduce there may be extra dims in the mask and it is not
```
Comment (nvaytet) [inline src/esssans/i_of_q.py:298]:
```suggestion
    # When not all dims are reduced there may be extra dims in the mask and it is not
```

**Review by nvaytet — APPROVED** [review]
Works like a charm :+1:

**Comment by nvaytet** [comment]
Can we also incorporate this change at the same time which keeps all the logical dimensions?
https://github.com/scipp/esssans/pull/57/files#diff-4deee399edfbe1368eec3c758e019a47f2972d9ebecb15878be74a16451f6234

**Comment by SimonHeybrock** [comment]
> Can we also incorporate this change at the same time which keeps all the logical dimensions? https://github.com/scipp/esssans/pull/57/files#diff-4deee399edfbe1368eec3c758e019a47f2972d9ebecb15878be74a16451f6234

Not sure which part your are linking to, just the removal of the flattening?

**Comment by nvaytet** [comment]
> Not sure which part your are linking to, just the removal of the flattening?

Yes, basically to keep the dimensions as `layer=4, tube=32, straw=7, pixel=512`.

**Pull request merged**